jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py

```diff
@@ -69,8 +69,12 @@ def _on_viewer_data_changed(self, msg=None):
         """
         viewer = self.app.get_viewer('spectrum-viewer')
 
-        self._spectral_subsets = self.app.get_subsets_from_viewer("spectrum-viewer",
-                                                                  subset_type="spectral")
+        try:
+            self._spectral_subsets = self.app.get_subsets_from_viewer("spectrum-viewer",
+                                                                       subset_type="spectral")
+        except ValueError:
+            pass
+
         self.spectral_subset_items = ["Entire Spectrum"] + sorted(self._spectral_subsets.keys())
```

Comment:
won't you still get an error raised here (`self._spectral_subsets.keys()`) if entering the except above?

Reply:
You should always have an empty `dict` at least from when the first data was loaded. Unless someone manages to define a subset before any data is loaded in the app I guess...